tools/tidy/src/synthesis/RegisterHasNoReset.cpp

```diff
@@ -95,6 +95,14 @@ struct MainVisitor : public TidyVisitor, ASTVisitor<MainVisitor, true, true> {
         if (symbol.drivers().empty())
             return;
 
+        // Skip iterator variables
+        if (symbol.kind == SymbolKind::Iterator)
```

Comment:
This is not right. The for loop variables are never SymbolKind::Iterator. The actual condition you want is to skip any variables that have Automatic lifetime.